x-pack/plugins/fleet/server/services/package_policy.ts

```diff
@@ -1111,7 +1112,9 @@ export function overridePackageInputs(
     }
 
     if (override.vars) {
-      originalInput = deepMergeVars(originalInput, override) as NewPackagePolicyInput;
+      const indexOfInput = inputs.indexOf(originalInput);
```

Comment:
Ahh I see the issue. Nasty bug - thank you for tracking it down.  We were mutating this object but never actually replacing it in the original array. Good catch!